hydrolib/core/dflowfm/xyn/name_validator.py

```diff
@@ -0,0 +1,62 @@
+class NameValidator:
+    """
+    Class used to validate the name from a given text.
+    Please note that this validator is specifically made for validating
+    names in .xyn files.
+    """
+
+    @staticmethod
+    def validate(name: str) -> str:
```

Comment:
This logic implies that the user cannot do the following:

```python
point.n = "this is a name"
```

but instead has to
```python
point.n = "'this is a name'"
```

after which the validator sets the name to the "originally wrong" one: `"this is a name"` without single quotes.
This is somewhat unexpected.
What if the user wants to copy the point to another point, then the validation of the second point will raise an error. But moreover, it is confusing behaviour when we say that example A is wrong, but then set it to example A ourselves.

The quotes are a requirement from the file and not the model. 
That is why I think we should handle it in the parser. After all, we also handle the quotes in the serializer.

Reply:
You are right! This is not working as expected. I will change the validation! I discussed the expected behavior with Jelmer and Arthur and updated the issue: https://github.com/Deltares/HYDROLIB-core/issues/459#issuecomment-1499996112